File: solutions/python/01-init/code/app/main.py
import sys

from dataclasses import dataclass

# import sqlparse - available if you need it!

database_file_path = sys.argv[1]
command = sys.argv[2]

if command == ".dbinfo":
    with open(database_file_path, "rb") as database_file:
        database_file.seek(16)  # Skip the first 16 bytes of the header
        page_size = int.from_bytes(database_file.read(2), byteorder="big")
        print(f"database page size: {page_size}")
else:
    print(f"Invalid command: {command}")

File: solutions/ruby/01-init/code/app/main.rb
database_file_path = ARGV[0]
command = ARGV[1]

if command == ".dbinfo"
  File.open(database_file_path, "rb") do |database_file|
    database_file.seek(16)  # Skip the first 16 bytes of the header
    page_size = database_file.read(2).unpack("n")[0]
    puts "database page size: #{page_size}"
  end
end
